dt-contacts/base-setup.php

```diff
@@ -127,13 +134,39 @@ public function dt_set_roles_and_permissions( $expected_roles ){
 
         $expected_roles["administrator"]["permissions"] = array_merge( $expected_roles["administrator"]["permissions"], $multiplier_permissions );
         $expected_roles["administrator"]["permissions"] = array_merge( $expected_roles["administrator"]["permissions"], $user_management_permissions );
+        $expected_roles['administrator']['permissions'] = array_merge( $expected_roles["administrator"]["permissions"], $role_caps );
         $expected_roles["administrator"]["permissions"]['manage_dt'] = true;
         $expected_roles["administrator"]["permissions"]['view_project_metrics'] = true;
         $expected_roles["administrator"]["permissions"]["dt_all_admin_contacts"] = true;
 
         return $expected_roles;
     }
 
+    public function dt_set_custom_roles_and_permissions( $roles ) {
```

Comment:
would this be better as part of the function in dt-core/setup-functions.php?

Reply:
@corsacca, I considered that, but that file is only run on admin requests. We need to register custom capabilities on the front end as well. Is there a better place I could put that code that would be available to the front-end as well?

Reply:
The code in dt-core/setup-functions.php only runs on wp-admin requests so that we aren't running the roles registering on every request.
Once registering is done, the roles settings persist across the all requests.
What part needs to be available on the front end?